### DISCUSSION TITLE
Adding packages with one-click apps

### DISCUSSION THREAD

**Discussion by 7aklhz** [caprover/caprover#1282] [Q&A] [answered]
When using one-click apps, how can I add packages during/before the installation process ? Concretely speaking, I'm installing mongodb using one-click apps and I would also like to have nano and wget installed on that container. In other words, how can I RUN something like apt-get install nano wget when installing one-click apps ? 

Is Pre-Deploy Script or Service Update Override the solution ? If so, how can I achieve this ?

**Comment by githubsaturn** [comment, marked as answer]
You'd have to "extend" the image. i.e.:
- Deploy the one click app as you normally would
- Once deployment is done, go to you app > deployment tab (3rd tab), and copy the name of the deployed image, in some cases it's simply something like `mongo:4`, in other cases it's something like `img-captain-mongodbapp:1`
- Now navigate to "Method 4: Deploy plain Dockerfile" and deploy this docker file
```Dockerfile
# replace this image, with your deployed image
FROM mongo:4 
RUN apt-get update && ..... 
```

**Reply by 7aklhz** [reply]
Hi again. It turns out I need to pass an extra command-line flags to the mongod service command and more precisely the following command line : --wiredTigerCacheSizeGB. 

The bitbamy docker image uses the following environmental variable to achieve this: 
MONGODB_EXTRA_FLAGS=--wiredTigerCacheSizeGB=2

I tried this however when I then check the maximum bytes configured (db.serverStatus().wiredTiger.cache), I still see the old figure. 

My question is: do you know how I can pass an extra command line flag ? 

Thanks

**Reply by githubsaturn** [reply]
You can either use the bitnami image or override the CMD line:
```Dockerfile
# replace this image, with your deployed image
FROM mongo:4 
RUN apt-get update && ..... 
CMD ["mongod" , "--someflag", "somevalue"]

```

**Reply by 7aklhz** [reply]
Great. Thanks. Works perfectly.

**Comment by 7aklhz** [comment]
works like a charm. Thanks
